File: frontend/src/api/parent.ts
// 家长端业务接口封装：对照方案 §6 Parent / §7 我的账户
import { get, post, http } from './client'

// 资料
export const getProfile = () => get<{ id: number; phone: string; nickname: string; avatar: string; status: number }>('/api/parent/profile')
export const updateProfile = (data: { nickname?: string; avatar?: string }) => post('/api/parent/profile', data)

// 孩子档案
export interface ChildItem {
  id: number
  parent_id: number
  name: string
  gender: number // 1=男 2=女
  birth_date: string
  remark: string
  first_visit: number
}
export const listChildren = () => get<ChildItem[]>('/api/parent/children')
export const createChild = (data: { name: string; gender: number; birth_date: string; remark?: string; first_visit?: number }) =>
  post<ChildItem>('/api/parent/children', data)
export const updateChild = (id: number, data: { name: string; gender: number; birth_date: string; remark?: string; first_visit?: number }) =>
  post<ChildItem>(`/api/parent/children/${id}`, data)
export const deleteChild = (id: number) => post(`/api/parent/children/${id}`, {})

// 我的预约
export interface ParentAppointment {
  id: number
  appointment_no: string
  store_name: string
  service_name: string
  child_name: string
  want_date: string
  want_slot: string
  status: string
  cancel_reason: string | null
  created_date: string
}
export const listAppointments = (params?: { status?: string; page?: number; page_size?: number }) =>
  get<{ items: ParentAppointment[]; total: number; page: number; page_size: number }>('/api/parent/appointments', params)
export const getAppointment = (id: number) => get<any>(`/api/parent/appointments/${id}`)
export const cancelAppointment = (id: number) => post(`/api/parent/appointments/${id}/cancel`, {})

// 兼容旧封装：部分页面直接引用 http
export { http }

File: api/app/schemas/parent.py
# -*- coding: utf-8 -*-
"""
家长端业务模型（app.schemas.parent）

功能说明：
- 我的资料 / 孩子档案 / 我的预约 的请求与响应模型（见 §6 Parent / §7.5 我的账户）。
- 孩子健康提醒按年龄生成（前台逻辑，schema 仅承载数据）。

依据：方案 §6 Parent 接口、§7 前台我的账户、§18 字段口径。
"""

from typing import List, Optional

from pydantic import BaseModel, ConfigDict


# ========== 孩子档案 ==========
class ChildIn(BaseModel):
    """新增/编辑孩子。前台我的孩子表单（见 §7 我的孩子）。"""

    name: str
    gender: int = 1  # 1=男 2=女
    birth_date: str  # YYYY-MM-DD
    remark: str = ""  # 过敏史等（仅后台可见）
    first_visit: int = 0  # 1=首诊 0=否


class ChildOut(ChildIn):
    """孩子档案响应（含 id 与归属家长）。"""

    model_config = ConfigDict(from_attributes=True)  # 允许从 ORM 实例直接校验（model_validate）

    id: int
    parent_id: int


# ========== 我的资料 ==========
class ParentProfileUpdateIn(BaseModel):
    """修改昵称/头像（见 §6 Parent profile PUT）。"""

    nickname: Optional[str] = None
    avatar: Optional[str] = None


# ========== 我的预约列表项 ==========
class ParentAppointmentOut(BaseModel):
    """我的预约列表项（见 §6 Parent appointments）。"""

    id: int
    appointment_no: str
    store_name: str = ""
    service_name: str = ""
    child_name: str = ""
    want_date: str = ""
    want_slot: str = ""
    status: str = "pending"
    cancel_reason: Optional[str] = None
    created_date: str = ""
